### DISCUSSION TITLE
Read-Only Rich Text Rendering

### DISCUSSION THREAD

**Discussion by RoboZoom** [sibiraj-s/ngx-editor#432] [Q&A] [answered]
Hi, I'm trying to render a read-only version of the component that retains the rich text formatting.  I did this by setting the `[contentEditable]` field to false:

```html
<div class="NgxEditor__Wrapper" *ngIf='canEdit'>
  <ngx-editor-menu [editor]="editor"> </ngx-editor-menu>
  <ngx-editor
    [editor]="editor"
    [formControl]='formControl'
    [placeholder]=placeholder
    [contentEditable]='canEdit'
  ></ngx-editor>
</div>
```

When I set `canEdit` to true, the component works great.  When I set it to `false`, nothing is rendered.

I created a stack blitz to replicate - I can't quite seem to exactly replicate, but the problem where the Text Box still allows editing persists: https://stackblitz.com/edit/ngx-editor-quickstart-48kags?file=src/app/app.component.ts

I'm sure that this is a common usage - am I doing something wrong here?

**Comment by sibiraj-s** [comment, marked as answer]
To disable the editor. just use the `disabled` property as you do for an input element. `<ngx-editor disabled></ngx-editor>`

**Reply by RoboZoom** [reply]
Thanks!

**Reply by henafufa** [reply]
@sibiraj-s I did add  disable but it is still editable

**Reply by henafufa** [reply]
any idea why? I will appreciate any help. thanks

**Reply by sibiraj-s** [reply]
The editor follows the behaviour that of a form input. Add the disabled property like how you treat a form input. Please share a reproducible example to investigate further.

**Reply by RoboZoom** [reply]
@henafufa - I found that the following solves the problem:

```

if(this.canEdit) this.formControl.enable();
else this.formControl.disable();
```

where in this case, the form control is the reactive form control assigned to the ngx-editor selector:

```
<div class="NgxEditor__Wrapper" [ngClass]="{'hide-style': hideStyle(canEdit)}">
  <ngx-editor-menu [editor]="editor" [toolbar]='getToolbar()' *ngIf='canEdit'> </ngx-editor-menu>
  <ngx-editor
    [editor]="editor"
    [formControl]='formControl'
    [placeholder]='placeholder'

  ></ngx-editor>
</div>
```

Using the disabled attribute like you'd do with a standard html selector does not work unfortunately.

**Reply by sibiraj-s** [reply]
Been not using angular lately, but for reactive forms disabled attributes won't work I guess. disabled is for using with ngModel.

What you are doing is correct, You can also pass the value and link it while creating the control as well.

something like

```
editor: [{value: '', disabled: this.isDisabled}],
```

**Reply by RoboZoom** [reply]
Understand - maybe it would be helpful to add some documentation in the readme about the difference in disabling a reactive versus non-reactive form control.

**Reply by sibiraj-s** [reply]
Yeah. I can add that. But this is really the angular way, the editor doesn't have any special behaviour so didn't bother much.

Will add in docs.

**Reply by RoboZoom** [reply]
I wrote it up and submitted a PR to try to be helpful.

**Reply by henafufa** [reply]
@RoboZoom @sibiraj-s  solved, thank you!

[a comment by an automated account is not shown]
